### DISCUSSION TITLE
[Question] : How to enforce type declaration for class, method , global variables ?

### DISCUSSION THREAD

**Discussion by ketannabera** [microsoft/pyright#4650] [General]
For below code snippet :

class MyClass:
   KEY = "key"
   
pyright config : 
[tool.pyright]
...
reportMissingImports = true
reportUnusedImport = true
reportUnusedVariable = true

pyright version : 1.1.293

 MyClass is included in the pyright check. 
 However, I want to enforce error about missing type info for KEY.
Can someone help with what config should I use to enforce that? 
Thanks !

**Comment by erictraut** [comment]
What information do you think is missing for `KEY`? Its type is `Literal["key"]` based on the value assigned to it. I suppose you could add an explicit type annotation for it, but why?

**Comment by erictraut** [comment]
I'm going to convert this to a discussion topic since it's a question, not a bug report.

**Reply by ketannabera** [reply]
So, I want it to enforce : 
KEY : str = "Key". Basically, want it to raise error with pyright checks if some type annotation is missing in code. 
What config should I do for the same ?
